### PR TITLE
Bwa 0.7.16

### PR DESCRIPTION
**Pull request by johanneskoester**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

I have also updated the homepage and added the two perl scripts.

### CODE REVIEW

**Comment by johanneskoester** [comment]
@bgruening there seems to be a problem with mulled: https://travis-ci.org/bioconda/bioconda-recipes/jobs/274969587#L826
It complains about an invalid bind argument...

**Comment by jtratner** [comment]
@johanneskoester - can you copy/template this over to the other bwa recipes too?

File: recipes/bwa/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,34 @@
...
+    - zlib {{CONDA_ZLIB}}*
   run:
-    - zlib
+    - zlib {{CONDA_ZLIB}}*
     - libgcc  # [not osx]
+    - perl
 
 test:
   commands:
     - bwa 2>&1 | grep 'index sequences in the'
```
Comment (jtratner) [inline recipes/bwa/meta.yaml]: @johanneskoester - this is causing mulled to fail (the `'` causes this command to break out of the shell escaping) - you need to use `"index sequences in the"` instead)

Comment (johanneskoester) [reply recipes/bwa/meta.yaml]: Thanks!

**Comment by johanneskoester** [comment]
@jtratner question is whether we have the need to update the old versions as well. Usually our policy is that we only modify the latest version unless somebody explicitly needs the older version (of course, the older package version won't be deleted). So, the fact that we have dedicated folders for the old versions here already indicates this. However, it could also be that they are not needed anymore.

* @pinguinkiste do you specifically need bwa 0.5.9 and 0.7.8?
* @JinfengChen do you specifically need bwa 0.6.2?

**Comment by jtratner** [comment]
@johanneskoester - gotcha - well if you could update 0.7.3a that'd be great :) . My understanding is that point versions of bwa are quite different (i.e., it doesn't use semantic versioning)

**Pull request merged**